roundabout/piece.js

```diff
@@ -0,0 +1,61 @@
+// NOTE: shim globals needed by content-claims client deps that would be present in nodejs v18.
+// TODO: migrate to sst v2 and nodejs v18+
+import './globals.js'
+
+import { read } from '@web3-storage/content-claims/client'
+import * as Raw from 'multiformats/codecs/raw'
+
+export const CAR_CODE = 0x02_02
+
+/** @see https://github.com/multiformats/multicodec/pull/331/files */
+export const PIECE_V2_CODE = 0x10_11
+
+/** 
+ * @typedef {import('multiformats/cid').Link} Link
+ * @typedef {import('@web3-storage/w3up-client/types').CARLink} CARLink
+ * @typedef {import('@web3-storage/content-claims/client/api').Claim} Claim
+ **/
+
+/**
+ * Return the cid if it is a Piece CID or undefined if not
+ *
+ * @param {Link} cid
+ */
+export function asPieceCid(cid) {
+  if (cid.multihash.code === PIECE_V2_CODE && cid.code === Raw.code) {
+    return cid
+  }
+}
+
+/**
+ * Return the cid if it is a CAR CID or undefined if not
+ *
+ * @param {Link} cid
+ */
+export function asCarCid(cid) {
+  if (cid.code === CAR_CODE) {
+    return cid
+  }
+}
+
+/**
+ * Find the set of CAR CIDs that are claimed to be equivalent to the Piece CID
+ * 
+ * @param {Link} piece
+ * @param {(Link) => Promise<Claim[]>} [fetchClaims]
+ */
+export async function findEquivalentCarCids (piece, fetchClaims = read) {
+  /** @type {Set<CARLink>} */
+  const cids = new Set()
+  const claims = await fetchClaims(piece)
+  for (const claim of claims) {
+    if (claim.type !== 'assert/equals') {
+      continue
+    }
+    const carCid = asCarCid(claim.content) ?? asCarCid(claim.equals)
```

Comment:
I'm bit lost here, I was expecting that `claim.content` will be set to `piece` here, although I might be misinterpreting what `fetchClaims` supposed to do.

That is to say I would appreciate code comment above this line telling me what's going on so reader does not has to guess. It also may be worth noting what are the expectations in regards to `fetchClaims` function.